pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -161,6 +161,25 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	return nil
 }
 
+// Test runs the devfile test command
+func (a Adapter) Test(testcmd versionsCommon.DevfileCommand, show bool) (err error) {
+	componentExists := utils.ComponentExists(a.Client, a.ComponentName)
+	if !componentExists {
+		return fmt.Errorf("component does not exist, a valid component is required to run 'odo test'")
+	}
```

Comment:
Do we really need this call since we are getting the pod just after it? Won't that be enough?

Reply:
This checks if the deployment exist; and we should show different error message if there is a problem when getting pod.

Reply:
But on a slow cluster both calls will cause performance problems. Since without the pod, it won't run anyways, just check for the pod and error out.

Reply:
Since this PR does both docker and kube; docker logic cannot be applied with the above logic since an odo component in docker can have multiple containers. So if a devfile has 2 components, that translates to 2 containers in docker. But if one of the containers is down; we cannot say a component exist with at least one container in our example.

I can honestly see where your argument is but i think even `odo watch` and `odo push` have these logic which can be refactored more optimally.

Reply:
> But if one of the containers is down; we cannot say a component exist with at least one container in our example.

For docker we can write it separately in the docker adapter. Logic in the Kubernetes should be independent of that.

Reply:
updated the Kubernetes logic